ompi/errhandler/errhandler_predefined.c

```diff
@@ -177,21 +177,30 @@ static void backend_fatal_aggregate(char *type,
                                     char *name, int *error_code,
                                     va_list arglist)
 {
-    char *arg, *prefix, *err_msg = "Unknown error";
-    bool err_msg_need_free = false;
+    char *arg = NULL, *prefix = NULL, *err_msg = NULL;
+    const char* const unknown_error_code = "Error code: %d (no associated error message)";
+    const char* const unknown_error = "Unknown error";
 
     arg = va_arg(arglist, char*);
     va_end(arglist);
 
-    asprintf(&prefix, "[%s:%d]", ompi_process_info.nodename,
-             (int) ompi_process_info.pid);
+    if (asprintf(&prefix, "[%s:%d]",
+                 ompi_process_info.nodename,
+                 (int) ompi_process_info.pid) == -1) {
+        // non-fatal, we could still go on to give useful information here...
+        opal_output(0, "%s", "Could not write node and PID to prefix");
```

Comment:
FWIW, I think compilers only warn on that for libc functions (e.g., printf).  They don't for opal_output.  But it doesn't matter that much...